libs/wingsdk/src/simulator/lockfile.ts

```diff
@@ -0,0 +1,125 @@
+import type { PathLike } from "fs";
+import fs, { type FileHandle } from "fs/promises";
+import path from "path";
+import { isNodeError } from "./util.js";
+
+// eslint-disable-next-line @typescript-eslint/no-shadow
+const exists = async (path: PathLike) => {
+  try {
+    await fs.stat(path);
+    return true;
+  } catch {
+    return false;
+  }
+};
+
+const UPDATE_INTERVAL = 1000;
+const STALE_THRESHOLD = 5000;
```

Comment:
(optional) Add a short description for each?

Reply:
Done.